src/core/state.go

```diff
@@ -578,7 +579,10 @@ func (state *BuildState) ExpandOriginalTargets() BuildLabels {
 // ExpandOriginalLabels expands any pseudo-labels (ie. :all, ... has already been resolved to a bunch :all targets)
 // from the set of original labels.
 func (state *BuildState) ExpandOriginalLabels() BuildLabels {
-	return state.ExpandLabels(state.OriginalTargets)
+	state.progress.originalTargetMutex.Lock()
+	targets := state.progress.originalTargets[:]
+	state.progress.originalTargetMutex.Unlock()
```

Comment:
nit: could use defer here

Reply:
I don't want to hold the lock around ExpandLabels though